### DISCUSSION TITLE
I am unable to upload images after logging in with auth.

### DISCUSSION THREAD

**Discussion by ru88s** [supabase/supabase#5922] [Questions] [answered]
I can't upload images after auth google login, it returns 400 error.
I can upload if you are not logged in.

I think it has something to do with Storage policies.
How do I write to allow image uploads from both un-logged in and logged in users

![スクリーンショット 2022-03-15 10 12 14](https://user-images.githubusercontent.com/42270511/158286439-68dcd1d2-fc40-41f6-9d4c-be45903fb8b5.png)

USING expression
```js
((bucket_id = 'images'::text) AND ((storage.foldername(name))[1] = 'public'::text))
```

**Comment by GaryAustin1** [comment, marked as answer]
You say upload, but show read (select) using Policy.  Do you have a policy for inserts and a use check policy for it?
Or did you mean download is not working?

**Reply by ru88s** [reply]
@GaryAustin1 
![スクリーンショット 2022-03-15 10 27 34](https://user-images.githubusercontent.com/42270511/158287853-4ba80306-a678-4eb3-af98-78297ca8a1ef.png)

Thank you for your reply.
I added a check for all.
But when I upload it while logged in, I get an error.

![スクリーンショット 2022-03-15 10 28 56](https://user-images.githubusercontent.com/42270511/158288016-b4f4d0b5-e6ac-459f-a5fa-fc43be66934b.png)

**Reply by GaryAustin1** [reply]
Can you show your insert policy?   It needs to have a "with check" policy.

**Reply by ru88s** [reply]
![スクリーンショット 2022-03-15 10 42 35](https://user-images.githubusercontent.com/42270511/158289238-8bfc3131-8e7e-4fc9-b41c-69b91102f08e.png)

Is this it?

The code I am doing is this.
```js
const { data } = await supabase.storage.from("images").upload(`${filename}`, file);
```

**Reply by GaryAustin1** [reply]
So you have a check on the folder (path) name including 'public/', but you are uploading a file to '/'.  Your policy wants path to contain 'public/' then the file name.... or get rid of the foldername part of the policy.

**Reply by ru88s** [reply]
![スクリーンショット 2022-03-15 11 01 29](https://user-images.githubusercontent.com/42270511/158291116-64c94a3a-0cfd-4d0e-a227-63dbe7ca336f.png)

I see.
`storage.foldername(name))[1] = 'public'::text`
was unnecessary.
I removed it and was able to upload the file successfully.
Thank you very much.
